File: FrontEnd/src/app/routes/os/closure/os-closure.models.ts
export type OsClosureGuardStatus = 'OK' | 'ALERTA' | 'BLOQUEIO';

export interface OsClosureGuard {
  code: string;
  label: string;
  status: OsClosureGuardStatus;
  message: string;
  owner: string;
  route?: string | null;
  overrideAllowed: boolean;
}

export interface OsClosureReviewModel {
  ordemServicoId: number;
  numeroOS: string;
  aggregateVersion: number;
  operationalState: 'EM_REVISAO' | 'PENDENCIA_CORRECAO' | 'CONCLUIDA_OPERACIONAL' | string;
  readyToComplete: boolean;
  alreadyCompleted: boolean;
  guards: OsClosureGuard[];
  partialSources: string[];
  snapshotId?: number | null;
  operationallyCompletedAt?: string | null;
}

export interface OsClosureCommandResult {
  ordemServicoId: number;
  numeroOS: string;
  operationalState: string;
  aggregateVersion: number;
  snapshotId: number;
  operationallyCompletedAt: string;
  downstreamRequestId?: string | null;
}

File: backend/src/main/java/com/neritech/saas/ordemservico/dto/OSClosureModels.java
package com.neritech.saas.ordemservico.dto;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Contratos canônicos da revisão e conclusão operacional da Ordem de Serviço.
 *
 * A conclusão operacional não gera contas a receber, fatura ou documento fiscal.
 * Esses efeitos pertencem ao comando separado de liberação para faturamento.
 */
public final class OSClosureModels {

    private OSClosureModels() {
    }

    public record Guard(
            String code,
            String label,
            String status,
            String message,
            String owner,
            String route,
            boolean overrideAllowed) {
    }

    public record Review(
            Long ordemServicoId,
            String numeroOS,
            Integer aggregateVersion,
            String operationalState,
            boolean readyToComplete,
            boolean alreadyCompleted,
            List<Guard> guards,
            List<String> partialSources,
            Long snapshotId,
            LocalDateTime operationallyCompletedAt) {
    }

    public record CommandResult(
            Long ordemServicoId,
            String numeroOS,
            String operationalState,
            Integer aggregateVersion,
            Long snapshotId,
            LocalDateTime operationallyCompletedAt,
            String downstreamRequestId) {
    }
}
